cpu/sam0_common/ldscripts/sam0_common.ld

```diff
@@ -21,10 +21,12 @@
  * @}
  */
 
+boot_offset = DEFINED( offset ) ? offset : 0x0 ;
+
 MEMORY
 {
-    rom (rx)    : ORIGIN = 0x00002000, LENGTH = 256K-0x2000
-    ram (rwx)   : ORIGIN = 0x20000000, LENGTH = 32K
+    rom (rx)    : ORIGIN = 0x00000000 + boot_offset, LENGTH = rom_length
```

Comment:
If you subtract boot offset from the length you will have one less source of mistakes in the makefiles.

Reply:
Yes! I thought about it but wanted to make it explicit... maye is not necessary so I'll change with your suggestion.